tools/parameter_tuning.py

```diff
@@ -0,0 +1,47 @@
+from sisyphus import Job, Task, tk
+from typing import Any, List, Tuple
+
+import numpy as np
+
+from i6_core.util import instanciate_delayed
+
+
+class PickOptimalParametersJob(Job):
+    """
+    Pick a set of optimal pickleable parameters based on their assigned (dynamic) score value.
+    """
+
+    def __init__(self, parameters: List[Tuple[Any]], values: List[tk.Variable], mode="minimize"):
+        """
+        :param parameters: list of tuples of parameters, must be pickleable
```

Comment:
I'm not really getting the reasoning for the `tuple` structure of the parameters. Why shouldn't it be possible to pass a list of dicts, lists or whatever else the user prefers as long as it's pickleable? I would make this more generic. In this case you could get rid of the `self.num_parameters`, the length check and only define one output var instead of a list.

Reply:
If everything is pickled you can only unpickle it "within" the next job, but we want to handle each individual parameter in the manager itself.

Reply:
if you want that behavior, you can still add a dict as single entry in the (formerly tuple) sequence.